webapp/src/Service/DOMJudgeService.php

```diff
@@ -993,13 +998,50 @@ public function unblockJudgeTasksForProblem(int $probId): void
         }
     }
 
-    public function maybeCreateJudgeTasks(Judging $judging, int $priority = JudgeTask::PRIORITY_DEFAULT): void
+    public function unblockJudgeTasksForSubmission(string $submissionId): void
+    {
+        // These are all the judgings that don't have associated judgetasks yet. Check whether we unblocked them.
+        $judgings = $this->helperUnblockJudgeTasks()
+            ->join(Submission::class, 's', Join::WITH, 'j.submission = s.submitid')
+            ->andWhere('j.submission = :submissionid')
+            ->setParameter('submissionid', $submissionId)
+            ->getQuery()
+            ->getResult();
+        foreach ($judgings as $judging) {
+            $this->maybeCreateJudgeTasks($judging, JudgeTask::PRIORITY_DEFAULT, True);
+        }
+    }
+
+    public function unblockJudgeTasks(): void
+    {
+        // These are all the judgings that don't have associated judgetasks yet. Check whether we unblocked them.
+        $judgings = $this->helperUnblockJudgeTasks()
+            ->getQuery()
+            ->getResult();
+        foreach ($judgings as $judging) {
+            $this->maybeCreateJudgeTasks($judging);
+        }
+    }
+
+    public function maybeCreateJudgeTasks(Judging $judging, int $priority = JudgeTask::PRIORITY_DEFAULT, bool $manualRequest = False): void
     {
         $submission = $judging->getSubmission();
         $problem    = $submission->getContestProblem();
         $language   = $submission->getLanguage();
 
-        if (!$problem->getAllowJudge() || !$language->getAllowJudge()) {
+        $evalOnDemand = False;
+        // We have 2 cases, the problem picks the global value or the value is set.
+        if ( ((int)$problem->getLazyEvalResults() === 0 && $this->config->get('lazy_eval_results') === static::EVAL_DEMAND)
```

Comment:
instead of 0 and 3 can we use human readable constants here?

Reply:
Yes, I thought I've done all the cases but apparently I had forgotten those.